antelope_contracts/contracts/erc20/src/erc20.cpp

```diff
@@ -247,16 +267,18 @@ void erc20::transfer(eosio::name from, eosio::name to, eosio::asset quantity,
     auto index = token_table.get_index<"by.symbol"_n>();
     auto itr = index.find(v);
 
-    eosio::check(itr != index.end() && itr->min_deposit.symbol == quantity.symbol, "received unregistered token");
-    eosio::check(quantity.amount >= itr->min_deposit.amount && quantity.amount > itr->deposit_fee.amount, "deposit amount too less");
+    eosio::check(itr != index.end() && itr->ingress_fee.symbol == quantity.symbol, "received unregistered token");
+    eosio::check(quantity.amount > itr->ingress_fee.amount, "deposit amount must be greater than ingress fee");
 
-    quantity.amount -= itr->deposit_fee.amount;
+    uint64_t ingress_fee = itr->ingress_fee.amount;
+    quantity.amount -= ingress_fee;
```

Comment:
Why are you bypassing the symbol checks? We want to make sure the symbols (with precision) are correct or abort the action. Because the symbol code was used to look up the table, we know the symbol codes match. But for some reason the precision may not. So better be defensive and check that.

Reply:
fixed. sorry i was not aware operator+ in eosio::asset